src/types.ts

```diff
@@ -216,6 +217,11 @@ export interface CommitData {
   committer?: UserData;
 }
 
+export interface CommitDataWithRequiredDate extends CommitData {
+  author?: Required<UserData>;
+  committer?: Required<UserData>;
+}
+
 export interface CommitSigner {
-  generateSignature(commit: CommitData): Promise<string>;
+  generateSignature(commit: CommitDataWithRequiredDate): Promise<string>;
```

Comment:
This seems like a breaking change which makes additional inputs required. Does this actually need to be a required field?